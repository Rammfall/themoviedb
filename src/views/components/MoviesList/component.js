import React from 'react'
import { Col, Row } from 'antd'
import PropTypes from 'prop-types'

import SearchForm from 'Views/components/SearchForm'
import MovieCard from 'Views/components/shared/MovieCard'
import PrivateLayout from 'Views/layouts/PrivateLayout'
import Pagination from 'Views/components/Pagination'
import MoviesLoader from 'Views/components/MoviesLoader'

const MoviesList = ({ movies, quantity, isLoading, current, getMovies }) => (
  <PrivateLayout>
    <>
      <SearchForm />
      <div className='top-margin'>
        {
          isLoading
            ? (
              <MoviesLoader />
            )
            : (
              <>
                <Row
                  type='flex'
                  gutter={16}
                >
                  <Col
                    className='gutter-row'
                    span={20}
                    offset={2}
                  >
                    <Row
                      gutter={{
                        xs: 8,
                        sm: 16,
                        md: 24,
                        lg: 32
                      }}
                    >
                      {movies.map(({ original_title: originalTitle, id, overview, poster_path: posterPath }) => (
                        <Col
                          key={id}
                          xs={{ span: 24 }}
                          sm={{ span: 12 }}
                          md={{ span: 8 }}
                          lg={{ span: 8 }}
                          xl={{ span: 6 }}
                        >
                          <MovieCard
                            title={originalTitle}
                            description={overview}
                            image={posterPath}
                          />
                        </Col>
                      ))}
                    </Row>
                  </Col>
                </Row>
                <Pagination
                  total={quantity}
                  current={current}
                  onChange={getMovies}
                />
              </>
            )
        }
      </div>
    </>
  </PrivateLayout>
)

MoviesList.propTypes = {
  movies: PropTypes.arrayOf(PropTypes.shape({
    original_title: PropTypes.string,
    id: PropTypes.number,
    overview: PropTypes.string,
    poster_path: PropTypes.string
  })).isRequired,
  quantity: PropTypes.number,
  isLoading: PropTypes.bool,
  current: PropTypes.number.isRequired,
  getMovies: PropTypes.func.isRequired
}

MoviesList.defaultProps = {
  isLoading: null,
  quantity: null
}

export default MoviesList
