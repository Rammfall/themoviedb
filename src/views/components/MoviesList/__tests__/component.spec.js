import React from 'react'
import { shallow } from 'enzyme'

import MoviesList from '../component'

describe('Pagination', () => {
  const defaultProps = {
    current: 1,
    movies: [{
      original_title: 'test',
      id: 2,
      overview: 'test',
      poster_path: 'test'
    }],
    quantity: 20,
    isLoading: false,
    getMovies: jest.fn()
  }

  describe('with default props', () => {
    it('matches snapshot', () => {
      const component = shallow(<MoviesList {...defaultProps} />)

      expect(component).toMatchSnapshot()
    })
  })

  describe('when isLoading', () => {
    it('matches snapshot', () => {
      const props = {
        ...defaultProps,
        isLoading: true
      }
      const component = shallow(<MoviesList {...props} />)

      expect(component).toMatchSnapshot()
    })
  })
})
