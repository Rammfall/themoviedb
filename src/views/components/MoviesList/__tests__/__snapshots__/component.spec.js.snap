// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination when isLoading matches snapshot 1`] = `
<PrivateLayout>
  <SearchForm />
  <div
    className="top-margin"
  >
    <MoviesLoader />
  </div>
</PrivateLayout>
`;

exports[`Pagination with default props matches snapshot 1`] = `
<PrivateLayout>
  <SearchForm />
  <div
    className="top-margin"
  >
    <Row
      gutter={16}
      type="flex"
    >
      <Col
        className="gutter-row"
        offset={2}
        span={20}
      >
        <Row
          gutter={
            Object {
              "lg": 32,
              "md": 24,
              "sm": 16,
              "xs": 8,
            }
          }
        >
          <Col
            key="2"
            lg={
              Object {
                "span": 8,
              }
            }
            md={
              Object {
                "span": 8,
              }
            }
            sm={
              Object {
                "span": 12,
              }
            }
            xl={
              Object {
                "span": 6,
              }
            }
            xs={
              Object {
                "span": 24,
              }
            }
          >
            <MovieCard
              description="test"
              image="test"
              title="test"
            />
          </Col>
        </Row>
      </Col>
    </Row>
    <Pagination
      current={1}
      onChange={[MockFunction]}
      total={20}
    />
  </div>
</PrivateLayout>
`;
